src/coreclr/tools/aot/Mono.Linker.Tests.Cases/RequiresCapability/RequiresInCompilerGeneratedCode.cs

```diff
@@ -0,0 +1,2287 @@
+			}
+		}
+
+		class StateMachinesOnlyReferencedViaReflection
+		{
+			[RequiresUnreferencedCode ("--TestIteratorOnlyReferencedViaReflectionWhichShouldSuppress--")]
+			[RequiresAssemblyFiles ("--TestIteratorOnlyReferencedViaReflectionWhichShouldSuppress--")]
+			[RequiresDynamicCode ("--TestIteratorOnlyReferencedViaReflectionWhichShouldSuppress--")]
+			static IEnumerable<int> TestIteratorOnlyReferencedViaReflectionWhichShouldSuppress ()
+			{
+				yield return 0;
+				MethodWithRequires ();
+			}
+
+			[RequiresUnreferencedCode ("--TestAsyncOnlyReferencedViaReflectionWhichShouldSuppress--")]
+			[RequiresAssemblyFiles ("--TestAsyncOnlyReferencedViaReflectionWhichShouldSuppress--")]
+			[RequiresDynamicCode ("--TestAsyncOnlyReferencedViaReflectionWhichShouldSuppress--")]
+			static async void TestAsyncOnlyReferencedViaReflectionWhichShouldSuppress ()
+			{
+				await MethodAsync ();
+				MethodWithRequires ();
+			}
+
+			[ExpectedWarning ("IL2026", CompilerGeneratedCode = true)]
+			[ExpectedWarning ("IL3002", ProducedBy = ProducedBy.Analyzer | ProducedBy.NativeAot, CompilerGeneratedCode = true)]
+			[ExpectedWarning ("IL3050", ProducedBy = ProducedBy.Analyzer | ProducedBy.NativeAot, CompilerGeneratedCode = true)]
+			static IEnumerable<int> TestIteratorOnlyReferencedViaReflectionWhichShouldWarn ()
+			{
+				yield return 0;
+				MethodWithRequires ();
+			}
+
+			[ExpectedWarning ("IL2026", CompilerGeneratedCode = true)]
+			[ExpectedWarning ("IL3002", ProducedBy = ProducedBy.Analyzer | ProducedBy.NativeAot, CompilerGeneratedCode = true)]
+			[ExpectedWarning ("IL3050", ProducedBy = ProducedBy.Analyzer | ProducedBy.NativeAot, CompilerGeneratedCode = true)]
+			static async void TestAsyncOnlyReferencedViaReflectionWhichShouldWarn ()
+			{
+				await MethodAsync ();
+				MethodWithRequires ();
+			}
+
+			[ExpectedWarning ("IL2026", "--TestIteratorOnlyReferencedViaReflectionWhichShouldSuppress--")]
+			[ExpectedWarning ("IL2026", "--TestAsyncOnlyReferencedViaReflectionWhichShouldSuppress--")]
+			// Analyzer doesn't emit additional warnings about reflection access to the compiler-generated
+			// state machine members.
+			[ExpectedWarning ("IL2026", "--TestIteratorOnlyReferencedViaReflectionWhichShouldSuppress--", ProducedBy = ProducedBy.Trimmer)]
+#if !RELEASE
+			[ExpectedWarning ("IL2026", "--TestAsyncOnlyReferencedViaReflectionWhichShouldSuppress--", ProducedBy = ProducedBy.Trimmer)]
+			// In debug mode, the async state machine is a class with a constructor, so a warning is emitted for the constructor.
+			// The MoveNext method is virtual, so doesn't warn either way.
+#else
+			// In release mode, the async state machine is a struct which doesn't have a constructor, so no warning is emitted.
+#endif
+			// Linker warns about reflection access to compiler-generated state machine members.
+			// TODO: implement this for NativeAot (https://github.com/dotnet/runtime/issues/68786)
```

Comment:
If the work is tracked by the issue I would remove the "TODO", just leave the link to the bug